scraper.rb

```diff
@@ -11,38 +11,30 @@
 OpenURI::Cache.cache_path = '.cache'
 
 def noko_for(url)
-  Nokogiri::HTML(open(url).read) 
+  Nokogiri::HTML(open(url).read)
 end
 
 def scrape_list(url)
   noko = noko_for(url)
-  noko.css('section#comments table tr a[href*="search"]/@href').map(&:text).uniq.each do |link|
+  noko.css('.table-responsive.table-striped tr a[href*="search"]/@href').map(&:text).uniq.each do |link|
     mp_url = URI.join url, link
     mp = noko_for(mp_url)
-    box = mp.css('section#main')
 
-    data = { 
-      id: mp_url.to_s[/search.php\?id=(.*)/, 1],
-      name: box.css('h2').text.gsub('Hon. ', '').strip,
-      constituency: box.xpath('.//h3[contains(., "District Representative")]/following-sibling::h3[1]').text,
-      party: box.xpath('.//h3[contains(., "Party List")]').text.gsub(/Party List\s*-\s*/, ''),
-      image: box.css('img[src*="images/16th"]/@src').text,
-      term: 16,
-      source: mp_url.to_s,
+    data = {
+      id:           mp_url.to_s[/search.php\?id=(.*)/, 1],
+      name:         mp.css('.text-primary').text.gsub('Hon. ', '').strip,
+      constituency: mp.xpath('//*[contains(@class,"text-primary")]/following::small[1][contains(text(),"District Representative")]').text,
```

Comment:
It seems weird to capture the 'District Representative' text here, only to substitute it back out again later, when they're already in different nodes.

```ruby
mp.xpath('.//text()[.="District Representative"]//following-sibling::text()')
```

might be simpler, and potentially more robust here, and avoid that substitution.

(With a similar approach for the Party info too)